src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -227,6 +230,19 @@ ProjectItemInstance projectReference
             // The properties on the project reference supersede the ones from the MSBuild task instead of appending.
             if (newProperties.Count == 0)
             {
+                if(ConversionUtilities.ConvertStringToBool(projectReference.Project.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName)) && String.IsNullOrEmpty(projectReference.GetMetadataValue(SetPlatformMetadataName)))
+                {
+                    projectReference.SetMetadata(SetPlatformMetadataName, GetCompatiblePlatform2(projectReference));
+
+                    var ReferencedProject = new Project(projectReference.EvaluatedInclude);
+                    var projectReferencePlatformMetadata = ReferencedProject.GetPropertyValue(PlatformMetadataName);
+                    var CurrentProjectPlatform = projectReference.Project.GetPropertyValue(PlatformMetadataName);
+                    var platformLookupTable = projectReference.Project.GetPropertyValue(PlatformLookupTableMetadataName);
+                    var referencedLookupTableMetadata = ReferencedProject.GetPropertyValue(PlatformLookupTableMetadataName);
+
+
+                    String SelectedPlatform = PlatformNegotiation.GetPlatformString(projectReferencePlatformMetadata, referencedLookupTableMetadata, CurrentProjectPlatform, platformLookupTable, projectReference.EvaluatedInclude);
```

Comment:
use `string` instead of `String`

Reply:
ended up using var here to match convention